### ISSUE TITLE
Blank Page Issues on Kindle Colorsoft

### ISSUE THREAD

**Issue by BastiGit** [ciromattia/kcc#768]
Hello,

I have had the new Kindle Colorsoft e-reader for 2 days. When I try to convert mangas with KCC using the CS12 profile, I get a white blank page every few pages. When I go back to the main menu and reopen the manga, the page then loads correctly instead of being blank.

However, this is very annoying as I have to repeat this every few pages. I didn't have this problem with my previous Kindle Paperwhite 7. Is there a workaround for this?

Thank you in advance for the response!

![grafik](https://github.com/user-attachments/assets/2ee68e61-247f-4211-90d0-e48ff872e521)

**Comment by axu2** [comment]
That's concerning. Maybe the colorsoft has a bug?

Try normal jpg. 

Try no upscale.

**Comment by BastiGit** [comment]
I have tried JPG, no upscale and PNG. Unfortunately, this problem also occurs here. 

However, if I turn the pages very slowly, it seems to occur only rarely.

**Comment by Antheology** [comment]
> Ideas: uncheck color mode
> 
> Or
> 
> Use scribe profile with color checked.

I am also having this issue I will try these suggestions.

**Comment by axu2** [comment]
@utopiafallen Do you have any insight into this issue?

**Comment by utopiafallen** [comment]
> However, if I turn the pages very slowly, it seems to occur only rarely.

This has been my experience as well. Other than that, sorry, I don't have much to add.

**Comment by dusk-exhale-paging** [comment]
Jumping on here to say that I am also experiencing this issue. Normally, I turn the Kindle off and back on to be able to see the pages or I get out of the book and get back in. I’ve been using KCC for a long time now with my paper white and just recently got my color soft and started to experience this blank page issue

**Comment by axu2** [comment]
Can anyone test if issue happens with humble bundle epubs sent via Send to Kindle?

If it doesn't happen something is wrong with kcc epub formatting. 

If it does then it's a colorsoft bug that amazon is unlikely to fix since kcc uses the old kf8 comic format.

**Comment by axu2** [comment]
I meant sending an EPUB from humble bundle directly via Send to Kindle, not using KCC at all.

**Comment by axu2** [comment]
To obtain a test file, you could pay $1 for a current humble bundle:

https://www.humblebundle.com/books/pathfinder-starfinder-comics-dynamite-books

**Comment by JuanPardos** [comment]
Experiencing this bug as well. I'm giving up on tweaking with KCC settings, none of them seem to work. As last resort, I slide from the bottom to the center of the screen to correctly view the page (opening chapter view) though it's disturbing.
The other option is using EPUB -> AZW3 but then I get wide margins and don't know how to reduce them.

**Comment by JuanPardos** [comment]
Can confirm that sending EPUB trough Send To Kindle also has this issue.

**Comment by axu2** [comment]
I have a suspicion about this issue. 

Could you please tell me the general file size of the final problematic KCC output? I'm specifically looking for the filesizes of the individyal images inside the epub that KCC generates. 

A screenshot of the unzipped epub images folder is fine if i can see individual filesizes.

Potentially related: 

- #932

**Comment by JuanPardos** [comment]
Here you go, if you need anything more feel free to ask :)
Settings used, sent with "Send To Kindle"

![Image](https://github.com/user-attachments/assets/fcfc7c9a-d10c-4f03-a342-3a1feff06932)
![Image](https://github.com/user-attachments/assets/88204135-1886-43e0-ae4f-74d5583dbcc7)
![Image](https://github.com/user-attachments/assets/9fa89e6b-6bb5-4b97-a4ba-31e245003c83)
![Image](https://github.com/user-attachments/assets/b891b9e6-670c-421f-bd05-38b8a842689c)

**Comment by axu2** [comment]
I wonder if switching to 8 bit color instead of 24 bit color would help.

**Comment by JuanPardos** [comment]
I have bad news, even with color mode off (the images seem to be grayscale 8 bit now) the blank page issue its still here.

**Comment by axu2** [comment]
@JuanPardos 

I have another thing you can try.

Do:

1) use kcc 7.5.0 (just released)
2) scribe profile
3) try both color mode and non color
3) jpg unchecked
4) use a source file >1920 pixels in height so it triggers special rendering mode

And tell me if problem still exists

**Comment by jeffmcneill** [comment]
Make sure Embeded Style is ON in the footer settings. And set T/B margins to as small as possible. May need to set L/R margins smaller also.

**Comment by JuanPardos** [comment]
I am back. Some tests I did:

Source images were 800x11130, KCC 7.5.0. 

1. Scribe profile, jpeg unchecked, color **ON**, MOBI/AZW3 output, side loaded with USB Calibre -> Blank page issue still here.
2. Scribe profile, jpeg unchecked, color **OFF**, MOBI/AZW3 output, side loaded with USB Calibre -> Blank page also.
3. Scribe profile, jpeg unchecked, color **ON**, EPUB (200MB limit) output, sent with Send to Kindle Web-> No blank page issue (hurray!). However something weird happened. Note not all pages looks the same, don't know if this is the special rendering in action but it looks like 1 image on top left, two if there is enough space for second. In pages with only 1 image there is wide right and bottom margin, the first image starts at top left always.

Except for the special rendering mode this is almost perfect, it has no blank page issue, color and no (top left) fixed margins.
Will try more setups and configs once I have some spare free time.


Didn't try to set margins or Embedded Style. Last time I did some Calibre CSS tweaks the books become unreadable.

Output of 3.
![Image](https://github.com/user-attachments/assets/acea4f91-91f9-4a10-bd9f-9ca5e48038e8)

Settings used in 3.
![Image](https://github.com/user-attachments/assets/00088b96-cc0d-4812-a153-4e55089aacb9)

**Comment by axu2** [comment]
Ok don't convert or edit with calibre, that will break formatting. 

USB load by dropping the mobi directly into kindle documents folder. Don't involve calibre at all. 

May be worth going back to colorsoft profile to test if you used calibre previously.

**Comment by JuanPardos** [comment]
UPDATE: With these settings (note: colorsoft profile) and the same source I was able to skip 50~ pages with no blank pages.
No calibre this time, sideloaded to documents folder.

Can't tell if this time was the source used, new KCC version, some settings, no calibre...
Need to do more testing but this looks promising.

![Image](https://github.com/user-attachments/assets/1a9c56a9-d256-4b40-91cf-6e9785b45ec1)

**Comment by JuanPardos** [comment]
Oh no, the blank page is back.
I don't know what to say... this looks random.

**Comment by axu2** [comment]
In general, it seems to be a problem if you turn pages too fast on colorsoft

**Comment by axu2** [comment]
@JuanPardos feel free to try 7.5.1, it should be available in a few minutes with the fix for 

> However something weird happened. Note not all pages looks the same, don't know if this is the special rendering in action but it looks like 1 image on top left, two if there is enough space for second. In pages with only 1 image there is wide right and bottom margin, the first image starts at top left always.

If the scribe special rendering mode fixes the colorsoft issue I'll add it to colorsoft.

**Comment by JuanPardos** [comment]
This is the result of 7.5.1 Kindle Scribe profile + Color + Epub with Send To Kindle:

Not fully fixed as I still get wide margins, though now only 1 image per page which is nice. And still not blank pages with this format (Calibre reports as .kfx). Even when turning pages very fast.

Same pages as the other day plus one random, but all seem to have this huge margin.
I mean... if there is any way to reduce or delete these huge margins would be better than Colorsoft profile + MOBI which still has the blank page issue with the new version.

Starting to think if .kfx format would be better than MOBI, because at the end is the format I am getting after using Send To Kindle.

![Image](https://github.com/user-attachments/assets/5a414016-b837-4ede-ae4d-536a98c374fe)
![Image](https://github.com/user-attachments/assets/a8536ad9-6800-408d-9300-98e95260e569)
![Image](https://github.com/user-attachments/assets/e075cefb-19c3-4a62-a6a6-295d6b6361a3)

**Comment by JuanPardos** [comment]
UPDATE: This is the same configuration but Colorsoft profile:

At this point the only problem I find is the align. And the slowness of Send To Kindle of course...

![Image](https://github.com/user-attachments/assets/df486ccf-2aa7-4e66-8565-98a994fbf5f7)
![Image](https://github.com/user-attachments/assets/1f181698-b715-4cf8-9b84-37f03a41ba91)

**Comment by tailune** [comment]
I have the same problem with colorsoft, tried diferent settings and the blank page allways came back. but i found the following you can export your file with whattever setting from kcc in mobi/azw3 or epub then import it to calibre and in calibre reformat to KFX then errase the other format from calibre and send the KFX file to kindle, this method errased the blank page issue and reading was normal the only issue is no matter what changes you make in the calibre KFX formating it adds a small margin to the top and bottom if it doesnt bother you then stick with this method its super simple and it works. 

But to me i find it a little bit distracting so i continued the search for a way to delete the margins and luckily i found this method.
Basically you use Kindle create to create a kpf file and use CLI to convert  to kfx. by doing this the margins are set correcly and the blanc page issue is resolved.

Here you can find the detailed process: https://innocen.at/2022/02/creating-kfx-manga-for-kindle-side-loading/

By the way i start the process from step 2 converting to CBZ with KCC this is done so all the images files have a sequential name so they follow the chapter order.

**Comment by dusk-exhale-paging** [comment]
> I have the same problem with colorsoft, tried diferent settings and the blank page allways came back. but i found the following you can export your file with whattever setting from kcc in mobi/azw3 or epub then import it to calibre and in calibre reformat to KFX then errase the other format from calibre and send the KFX file to kindle, this method errased the blank page issue and reading was normal the only issue is no matter what changes you make in the calibre KFX formating it adds a small margin to the top and bottom if it doesnt bother you then stick with this method its super simple and it works.
> 
> But to me i find it a little bit distracting so i continued the search for a way to delete the margins and luckily i found this method. Basically you use Kindle create to create a kpf file and use CLI to convert to kfx. by doing this the margins are set correcly and the blanc page issue is resolved.
> 
> Here you can find the detailed process: https://innocen.at/2022/02/creating-kfx-manga-for-kindle-side-loading/
> 
> By the way i start the process from step 2 converting to CBZ with KCC this is done so all the images files have a sequential name so they follow the chapter order.

Yup, this is what we've already been doing and discussed over at the discord.

**Comment by axu2** [comment]
Ok, I copied some of the special scribe code to the colorsoft code path, MAYBE it works. Please test, I don't have a colorsoft myself. From past experience, JPG/PNG is likely the only setting that makes a difference.

And don't turn your pages TOO fast. Kindles always freak out if its too fast, but as long as a good reading pace doesn't blank out.

https://github.com/axu2/kcc/releases/tag/v8.0.4-colorsoft

@BastiGit @Antheology @dusk-exhale-paging @JuanPardos @tailune 

While you are at it, this also includes a new sharper rainbow eraser algorithm and enhanced pdf support including humble/fanatical that I'd love to hear your thoughts on.

**Comment by tailune** [comment]
> Ok, I copied some of the special scribe code to the colorsoft code path, MAYBE it works. Please test, I don't have a colorsoft myself. From past experience, JPG/PNG is likely the only setting that makes a difference.
> 
> And don't turn your pages TOO fast. Kindles always freak out if its too fast, but as long as a good reading pace doesn't blank out.
> 
> https://github.com/axu2/kcc/releases/tag/v8.0.4-colorsoft
> 
> [@BastiGit](https://github.com/BastiGit) [@Antheology](https://github.com/Antheology) [@dusk-exhale-paging](https://github.com/dusk-exhale-paging) [@JuanPardos](https://github.com/JuanPardos) [@tailune](https://github.com/tailune)
> 
> While you are at it, this also includes a new sharper rainbow eraser algorithm and enhanced pdf support including humble/fanatical that I'd love to hear your thoughts on.

I tested with the new build you sugested: if you leave upscaling unchecked and read at a slow pace the blank page almost never happens, it mostly happens with double spread pages and some times its not a completely blank page but it only renders de bottom 1/4 page, closing at a fully rendered page and re opening renders the blank pages and semi rendered pages correctly.

**Comment by JuanPardos** [comment]
I also noticed a slight decrease on blank pages with the new build for Colorsoft. Didn't test a lot of configs though.

**Comment by axu2** [comment]
Make sure you try both jpg/png. That makes the difference on scribe at least.

The upscale unchecked might matter...

We want to solve the problem of blank pages when reading at a normal pace. Faster than that will always blank out.

But feedback so far seems that the change doens't make much of a difference...

**Comment by ValadAmoleo** [comment]
Just got a colorsoft and transferred over Humble Bundle manga that I'd converted using KCC and I'm also getting this issue.  It's worth mentioning that on my Kindle Voyage the very same files do not have this issue.  I haven't reconverted them at all.

**Comment by RaiBoeira** [comment]
Got my kindle colorsoft recently and I'm having the same issue with the converted comics, any updates on a future fix update or something?
It doesn't seem to have a pattern on fast page turning or different settings of conversion, it seems really random for me.

**Comment by axu2** [comment]
Unfortunately I don't think Amazon will ever fix this.

Key points are

1) Colorsoft never supported the Download & Transfer feature, which enabled you to download and usb transfer manga AZW3 directly from the Kindle store. So this means that the AZW3 manga renderer was never tested on Colorsoft in Amazon internally and that they don't care it's currently broken.

I'd suggest just getting the 7" Kobo Libra Colour

**Comment by RaiBoeira** [comment]
Yeah, make sense, but the comics that are bought and downloaded directly from Amazon are not the same format? Couldn't be that KCC just hadn't figure it out what is the issue with the format that is converted?

**Comment by axu2** [comment]
Comics directly downloaded are KFX. KCC is kf8. 

The x is basically 10

**Comment by RaiBoeira** [comment]
Gotcha! Then I believe is not that easy to update the KCC to convert to KFX instead of KF8.
 (I'm not a coder, just a curious user by the way :)

**Comment by AcidWeb** [comment]
No idea about Colorsoft but older Kindles had the same issue and it was caused by page structure and incorrect size of output images.

**Comment by RaiBoeira** [comment]
So yeah, I was wondering if it was a KCC or a Kindle issue. Looks like is maybe both? Cause Amazon wouldn't fix a bug of a format that they don't use anymore and for the KCC could be hard to update to the newest format that Amazon uses.

I can live with this bug, just exit the comic and go back solves it. BUT i hope this gets fixed someday.

Thanks guys

**Comment by axu2** [comment]
There is no way currently to automate kfx creation of comic style books.

**Comment by axu2** [comment]
Please try this build where jpg quality was changed from 85 to 90:

https://github.com/axu2/kcc/releases/tag/v9.3.7-jpg90

This creates files that take less cpu time to decode but are larger. I suspect that a blank page occurs when the cpu times out during image decoding. Please use jpg unchecked.

@BastiGit @Antheology @utopiafallen @dusk-exhale-paging @JuanPardos @tailune @ValadAmoleo @RaiBoeira

**Comment by JuanPardos** [comment]
You may be right. Tested this build a bit and got a blank after 44 pages. That is while turning pages pretty fast.
Don't remember exactly but I think I was getting them more often. Can't tell for sure though, haven't been using my kindle lately...

**Comment by axu2** [comment]
@JuanPardos turning pages rapidly puts a lot of cpu strain, it will always happen that way. This hopefully prevents it from happening at normal reading pace

**Comment by RaiBoeira** [comment]
I messed around with the settings in the 9.1.0 version a few months ago and with a simple configuration I got the blank screen issue gone, in a comic of 130 pages using panel view all the time I didnt have the issue once. Using this settings I reconverted comics that previously had the issue and seems to work 99%. In another 130 page comic I had 1 blank screen in page 3 and never again, and it was on page view, when it zoom in with panel view it was visible.

Im using the Kindle Colorsoft, probably wont work in every device that had the issue but is worth the try. the settings are pretty simple , thats may be the reason that worked

(I also had a few comics with pages with no dialogs that resulted in fast page turning and had no issue)

<img width="1110" height="549" alt="Image" src="https://github.com/user-attachments/assets/bbf67877-158c-4d43-815e-a37c06c3d2a0" />

**Comment by neontheboss** [comment]
Pour moi le problème n'est pas résolu. Si j'utilise la dernière pre-release j'ai encore des pages blanches ou pire encore des demi pages. J'ai hâte que vous trouviez une solution

**Comment by axu2** [comment]
@neontheboss Unfortunate, so you are saying that 9.3.8 before the jpg quality change is better?

**Comment by neontheboss** [comment]
Il y en a moins mais malheureusement il y en a encore

**Comment by FlorentMartinPilot** [comment]
Not sure if it's the right place to post, as I'm using the first Scribe version, not the color one.

Even with the latest version, I have blank pages, or sometimes half-ones...

Even reverting to older versions, the problem is still present. Is it a bug in the Kindle software ?

Is there an older KCC version that could work ?

Thanks a lot for your great software anyway !

**Comment by axu2** [comment]
@FlorentMartinPilot 

Please make a new thread with your kcc settings. Png will cause blanks.

**Comment by neontheboss** [comment]
![Image](https://github.com/user-attachments/assets/48bdeee5-e628-4a71-8d0f-898b8cbb489b)
Avec  dernière version j’ai des pages comme ça en plus des pages blanches. 
J’ai même essayé avec la qualité des jpg à 95% mais ça n’a rien changé. 
Auriez-vous une idée?

**Comment by axu2** [comment]
@neontheboss 

What settings are you currently using?

Maybe try JPG, 85 quality, and disable cropping.

**Comment by neontheboss** [comment]
<img width="1817" height="1103" alt="Image" src="https://github.com/user-attachments/assets/06d80ada-9069-48d1-ad38-578ff539b46f" />
Voici mes réglages. Et j'ai les pages blanches en 85% et 95%.

**Comment by axu2** [comment]
Oh you are using Scribe Colorsoft. Yea that's not compatible at this time. Maybe disable cropping fixes it or just use pdf output @neontheboss

**Comment by neontheboss** [comment]
<img width="1794" height="519" alt="Image" src="https://github.com/user-attachments/assets/1198def5-99a6-495a-9fcd-feac8937ed63" />
Voici les meilleurs réglages que j'ai pour le moment. En pdf les marges sont trop grandes donc l'image est trop petite et les textes sont illisible mais par contre il n'y a plus de pages blanches. En mobi j'ai encore pas mal de pages blanche...

**Comment by crow1943** [comment]
I’ve heard that converting it to KFX format might improve things. Has anyone tried it? Currently, KCC doesn’t support KFX conversion, only EPUB.

**Comment by axu2** [comment]
@crow1943 You can use KCC to convert to PDF and usb transfer it, the latest firmware apparently supports fullscreen color for usb loading. If the update isn't activated for you yet, you can use Send to Kindle with the PDF.

**Comment by crow1943** [comment]
> [@crow1943](https://github.com/crow1943) You can use KCC to convert to PDF and usb transfer it, the latest firmware apparently supports fullscreen color for usb loading. If the update isn't activated for you yet, you can use Send to Kindle with the PDF.

Thanks for the heads-up! I tried what you suggested, and converting to PDF actually works on the latest version. The color looks great, and there are no more blank pages.

I also just tried converting to KFX format, and that works perfectly as well. I haven't run into any issues, and even the layout is working exactly as it should.

**Comment by axu2** [comment]
I recently learned that KCC had the wrong resolution for the Paperwhite 12th.

Is it wrong for colorsoft too? Having a wrong resolution might cause blank problems...

- #1299 

double tap any opposite corners to take a screenshot you can see via usb. you need to tap the very deep into the corners. then look at screenshot and tell me resolution.

**Comment by axu2** [comment]
Yup, it's confirmed to be wrong too. Will make a new release shortly.

**Comment by axu2** [comment]
If anyone here wants to confirm i released hotfix in 9.7.2

**Comment by axu2** [comment]
You can also use PDF output for Scribe 2025 series and colorsoft, latest update makes USB PDF fullscreen color.

**Comment by axu2** [comment]
Based on user reports the resolution change fixed it!
